### PR TITLE
chore: drop ZooKeeper 3.8.4

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

Part of #832.

- Remove ZooKeeper 3.8.4 (it was deprecated in SDP 24.7).
- Bump to Java 17 runtime for ZooKeeper 3.9.2 (still needs to be built with Java 11).

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
- [x] ~Changes are OpenShift compatible~ No change
- [x] ~All added packages (via microdnf or otherwise) have a comment on why they are added~ None added
- [x] ~Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there~ None added
- [x] ~All packages should have (if available) signatures/hashes verified~ None added
- [x] Add an entry to the CHANGELOG.md file
- [x] ~Integration tests ran successfully~ No new versions to test
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: zookeeper/versions.py [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-        "java-base": "11",
-        "java-devel": "11",
-        "jmx_exporter": "1.0.1",
-    },
     {
         "product": "3.9.2",
-        "java-base": "11",
-        "java-devel": "11",
+        "java-base": "11",  # Builds fail on Java 17, so it can stay as is.
+        "java-devel": "11",  # Builds fail on Java 17, so it can stay as is.
```
Comment (maltesander) [inline zookeeper/versions.py:5]: Should we put the "error" message or at least the package that failed in the comment for future ref?

**Review by maltesander — APPROVED** [review]
LGTM!

File: zookeeper/versions.py [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
-        "product": "3.8.4",
-        "java-base": "11",
-        "java-devel": "11",
-        "jmx_exporter": "1.0.1",
-    },
     {
         "product": "3.9.2",
-        "java-base": "11",
-        "java-devel": "11",
+        "java-base": "11",  # Builds fail on Java 17, so it can stay as is.
```
Comment (sbernauer) [inline zookeeper/versions.py:4]: You could try compiling it on 11 and running it on 17 (we did that in the past for other products)

**Review by sbernauer — APPROVED** [review]

File: zookeeper/versions.py (hunk 2 above)
Comment (NickLarsenNZ) [reply zookeeper/versions.py:4]: Good point, I'll give it a try.

File: zookeeper/versions.py (hunk 1 above)
Comment (NickLarsenNZ) [reply zookeeper/versions.py:5]: The error wasn't helpful at all, but I can mention the package.

File: zookeeper/versions.py (hunk 2 above)
Comment (NickLarsenNZ) [reply zookeeper/versions.py]: I tested based on b8487a3cca3c735201b789e2ea54845671d61ea6, and smoke-latest passed.

```bash
zookeeper-operator $ ./scripts/run-tests --test-suite smoke-latest
```

```
--- PASS: kuttl (106.60s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.9.2_use-server-tls-false_use-client-auth-tls-false_openshift-false (106.59s)
PASS
```

Would you consider this good enough?

File: zookeeper/versions.py (hunk 1 above)
Comment (NickLarsenNZ) [reply zookeeper/versions.py]: Fixed in b8487a3

**Review by Techassi — APPROVED** [review]

File: zookeeper/versions.py (hunk 2 above)
Comment (NickLarsenNZ) [reply zookeeper/versions.py]: I'll consider this resolved so we can move forward. Let me know if you don't think this is an adequate proof and we can work on it in a new PR.

Comment (sbernauer) [reply zookeeper/versions.py]: That works for me, thanks!

**Pull request merged**